### ISSUE TITLE
No matching toolchains with `experimental_repository_resolved_file`

### ISSUE THREAD

**Issue by keith** [bazelbuild/bazel#8523]
### Description of the problem / feature request:

When using the `experimental_repository_resolved_file` family of flags as recommended by [this post](https://blog.bazel.build/2018/09/28/first-class-resolved-file.html), when using rules_go you end up with this error upon building:

```
ERROR: While resolving toolchains for target //lang:go_default_library: no matching toolchains found for types @io_bazel_rules_go//go:toolchain
ERROR: Analysis of target '//lang:go_default_library' failed; build aborted: no matching toolchains found for types @io_bazel_rules_go//go:toolchain
```

Note: this is an easy repro case with go, but I've also seen this with other toolchains:

```
ERROR: While resolving toolchains for target //tools/storyboarder:storyboarder: no matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

- Clone https://github.com/bazelbuild/buildtools
- Run `bazel sync --experimental_repository_resolved_file=resolved.bzl`
- Run `bazel build --experimental_resolved_file_instead_of_workspace=resolved.bzl --experimental_repository_hash_file=resolved.bzl --experimental_verify_repository_rules=@bazel_tools//tools/build_defs/repo:git.bzl%git_repository --experimental_verify_repository_rules=@bazel_tools//tools/build_defs/repo:http.bzl%http_archive //...`

(or add these flags to a `.bazelrc`)

### What operating system are you running Bazel on?

macOS

### What's the output of `bazel info release`?

release 0.26.0

###  Have you found anything relevant by searching the web?

No, I also can't really tell what the state of these flags is

**Comment by aiuto** [comment]
Given to both configurability (because toolchains) and ExternalDeps. I am not sure where the boundary is.

**Comment by katre** [comment]
This looks suspiciously similar to https://github.com/bazelbuild/bazel/issues/7773, but I don't know what the `--experimental_repository_resolved_file` option is trying to do.

I suspect @aehlig may need to investigate this.

**Comment by aehlig** [comment]
> I suspect @aehlig may need to investigate this.

tl;dr: fixed upstream

The longer story is the following. The repository resolved file was once faithfully representing the semantics of the of the `WORKSPACE` file as a flat value. But then, the tool chains were introduced as a new concept but not added to the resolved file. It took me a while to realize that, and once I did, I fixed it in  384fc0bc2ae0ab084ff5a9e01bfdee17d123e017. That commit didn't make it into 0.26, but will be in 0.27 (as the resolved file still is officially "experimental", that commit didn't justify a cherry-pick or a patch release).

**Comment by katre** [comment]
Thank you! Is there a separate commit adding support for register_execution_platforms?

**Comment by aehlig** [comment]
> Thank you! Is there a separate commit adding support for register_execution_platforms?

Is that another feature added to `WORKSAPCE` files? Then I guess I'll have to add that to the resolved file as well.

**Comment by aehlig** [comment]
Pending for review: https://bazel-review.googlesource.com/c/bazel/+/102230
